Libraries/LibJS/AST.cpp

```diff
@@ -258,9 +258,12 @@ Value IfStatement::execute(Interpreter& interpreter, GlobalObject& global_object
 Value WhileStatement::execute(Interpreter& interpreter, GlobalObject& global_object) const
 {
     Value last_value = js_undefined();
-    while (m_test->execute(interpreter, global_object).to_boolean()) {
+    for (;;) {
+        auto test_result = m_test->execute(interpreter, global_object);
         if (interpreter.exception())
             return {};
```

Comment:
this apparently was an attempt to circumvent this very problem - but by the time we called `to_boolean()` on the empty value it was already too late.